### PR TITLE
fix deepkeys to work with empty arrays and objects

### LINKED ISSUE #104: deepKeys returns strange result
```deepKeys({ x: 1, y: "1", z: [1], zz: [], o: {}, n: null, u: undefined, nan: NaN })```

returns

```['x', 'y', 'z[0]', 'n', 'u', 'nan']```

Why is it missing `zz` and `o`, for me it is looks as bug

### PR DESCRIPTION
**Pull request by AuthorProxy**
fix #104

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,6 +3,10 @@ const isObject = value => {
 	return value !== null && (type === 'object' || type === 'function');
 };
 
+const isEmptyObject = value => {
+	return isObject(value) && Object.entries(value).length === 0;
```
Comment (sindresorhus) [inline index.js:7]:
```suggestion
	return isObject(value) && Object.keys(value).length === 0;
```

**Comment by sindresorhus** [comment]
CI is failing

**Review by mmkal — APPROVED** [review]
IMO: this behaviour is a little more intuitive than skipping empty objects. It'd be great to have something explicit in the docs though.

**Comment by sindresorhus** [comment]
You need to update the docs.

File: readme.md [hunk 2]
```diff
@@ -110,7 +110,7 @@ console.log(getProperty(object, escapedPath));
 
 ### deepKeys(object)
 
-Returns an array of every path. Plain objects are deeply recursed and are not themselves included.
+Returns an array of every path. Plain not empty objects are deeply recursed and are not themselves included. Empty objects or objects without values are included, e.g [], {}, null, undefined or NaN.
```
Comment (sindresorhus) [inline readme.md:113]: This is more confusing than it needs to be. Just say that non-empty plain objects and arrays are not themselves included.

**Comment by sindresorhus** [comment]
You also need to update index.d.ts docs.

**Pull request merged**